### PR TITLE
global: move internal ran functions to global scope

### PR DESCRIPTION
**Pull request by edcdavid**
Note: Now relies on https://github.com/openshift-kni/eco-goinfra/pull/560 to be merged first
Moving some functions previously defined in the RAN scope to global scope:

- ~ExecLocalCommand~
- ExecCmdWithRetries
- ExecCmdWithStdoutWithRetries
- ExecCommandOnSNOWithRetries
- isErrorExecuting
- WaitForClusterRecover
- IsPodHealthy
- IsPodInCondition
- SoftRebootSNO
- waitForClusterReachable
- waitForAllPodsHealthy

This is PR is split from https://github.com/openshift-kni/eco-gotests/pull/83 to review the global code only

### CODE REVIEW

File: tests/internal/params/vars.go
Comment (kononovn) [inline tests/internal/params/vars.go:16]: This parameter has to be added to https://github.com/openshift-kni/eco-gotests/blob/60d66789c0a6707c226d20d344dc1618ccb368ba/tests/internal/config/config.go#L32 instead of hardcode.

File: tests/internal/params/vars.go
Comment (kononovn) [inline tests/internal/params/vars.go:14]: It's OK to duplicate this LogLevel no need to move it to global vars.

File: tests/internal/cluster/cluster.go
Comment (kononovn) [inline tests/internal/cluster/cluster.go:191]: Please move this log message to the beginning of the function.

File: tests/internal/cluster/cluster.go
Comment (kononovn) [inline tests/internal/cluster/cluster.go:207]: It'll be good to add sleep interval as a argument.

File: tests/internal/cluster/cluster.go
Comment (kononovn) [inline tests/internal/cluster/cluster.go:223]: Please add glog message

File: tests/internal/cluster/cluster.go
Comment (kononovn) [inline tests/internal/cluster/cluster.go:228]: It'll be good to add sleep interval as a argument.

File: tests/internal/cluster/cluster.go
Comment (kononovn) [inline tests/internal/cluster/cluster.go:243]: Please add glog message.

File: tests/internal/cluster/cluster.go
Comment (kononovn) [inline tests/internal/cluster/cluster.go:260]: Please move private function to the end of the file.

File: tests/internal/cluster/cluster.go
Comment (kononovn) [inline tests/internal/cluster/cluster.go:281]: What is the point of this sleep ?

File: tests/internal/cluster/cluster.go
Comment (kononovn) [inline tests/internal/cluster/cluster.go:287]: This function can be removed. Please use eco-goinfra wait functions. You can add additional function is current functionality is not enough.

https://github.com/openshift-kni/eco-goinfra/blob/63369f7a7585eab062658272d3fd1b95b166b770/pkg/pod/pod.go#L274

File: tests/internal/cluster/cluster.go
Comment (kononovn) [inline tests/internal/cluster/cluster.go:306]: Same comment as above: https://github.com/openshift-kni/eco-gotests/pull/128/files#r1698723302

**Review by kononovn — CHANGES_REQUESTED** [review]
Please check few comments above.

**Comment by kononovn** [comment]
@edcdavid Please change PR name/commit message to something likeL
`global: move internal ran funcs to global scope`

File: tests/internal/cluster/cluster.go
Comment (klaskosk) [reply tests/internal/cluster/cluster.go:287]: I agree with Nikita here, there is room for these utilities to be added to eco-goinfra. It's been in the backlog for RAN to reorganize these helpers so they're not as good as they could be

**Comment by klaskosk** [comment]
From a RAN perspective, I'm hesitant to move all of these functions out of RAN's directory since as Nikita has noted there's plenty of room for improvement, much of it is specific to the team, and we have to be careful about not breaking CI with changes.

That being said, I'd like to hear @mcornea's thoughts on how system tests would approach these functions in case there's a different way that's more in line with what they're doing. Retries exist because CI would fail when the MCO daemon pod restarted in the middle of a test, so I'm not sure if they have encountered a similar problem.

If moving these functions really is the best way then I'd like to take the time to ensure it doesn't cause any automation issues for RAN but am not otherwise opposed

**Comment by mcornea** [comment]
> That being said, I'd like to hear @mcornea's thoughts on how system tests would approach these functions in case there's a different way that's more in line with what they're doing. Retries exist because CI would fail when the MCO daemon pod restarted in the middle of a test, so I'm not sure if they have encountered a similar problem.
> 
IMO we should take a staged approach by first adding these new functions globally and secondly refactor existing tests to use them. Since there's a risk the newly added functions introduce unexpected breakages to existing tests then this change needs to be properly planned according to each team's capacity. I'd personally first add the new global functions, then add the new system tests using the new functions and lastly refactor any of the existing ran or system-tests to make use of the new global functions.

**Comment by klaskosk** [comment]
> IMO we should take a staged approach by first adding these new functions globally and secondly refactor existing tests to use them.

This seems like a pretty fair approach. I had already set aside some time next week for the reorganization of ran helpers so I'll be able to dedicate some capacity then to getting these changes tested and implemented.

---

This came up when discussing refactoring inside of ran, but it may be worth considering naming the global package something more like command rather than cluster. Especially if we're moving `ExecLocalCommand` which doesn't do anything cluster related. I'll leave that up to @kononovn however.

File: tests/internal/params/vars.go
Comment (edcdavid) [reply tests/internal/params/vars.go]: fixed

File: tests/internal/cluster/cluster.go
Comment (edcdavid) [reply tests/internal/cluster/cluster.go]: fixed

File: tests/internal/cluster/cluster.go
Comment (edcdavid) [reply tests/internal/cluster/cluster.go]: fixed

File: tests/internal/cluster/cluster.go
Comment (edcdavid) [reply tests/internal/cluster/cluster.go:233]: fixed

File: tests/internal/cluster/cluster.go
Comment (edcdavid) [reply tests/internal/cluster/cluster.go]: fixed

File: tests/internal/cluster/cluster.go
Comment (edcdavid) [reply tests/internal/cluster/cluster.go]: fixed

File: tests/internal/cluster/cluster.go
Comment (edcdavid) [reply tests/internal/cluster/cluster.go]: fixed

File: tests/internal/cluster/cluster.go
Comment (edcdavid) [reply tests/internal/cluster/cluster.go]: This was to fix instability after reboot when running commands. This is no longer needed with retries. I removed it.

File: tests/internal/cluster/cluster.go
Comment (edcdavid) [reply tests/internal/cluster/cluster.go]: Fixed. Created https://github.com/openshift-kni/eco-goinfra/pull/560

File: tests/internal/cluster/cluster.go
Comment (edcdavid) [reply tests/internal/cluster/cluster.go]: fixed. created https://github.com/openshift-kni/eco-goinfra/pull/560

File: tests/internal/params/vars.go
Comment (edcdavid) [reply tests/internal/params/vars.go]: fixed

File: tests/internal/cluster/cluster.go
Comment (edcdavid) [reply tests/internal/cluster/cluster.go]: Fixed

File: tests/internal/cluster/cluster.go
Comment (edcdavid) [reply tests/internal/cluster/cluster.go]: Fixed.

File: tests/internal/cluster/cluster.go
Comment (edcdavid) [reply tests/internal/cluster/cluster.go]: Fixed.

File: tests/internal/params/vars.go
Comment (kononovn) [inline tests/internal/params/vars.go:4]: Why did you change it? Can you revert it back to it's original state?

File: tests/internal/config/default.yaml
Comment (kononovn) [inline tests/internal/config/default.yaml:14]: Please re-name to logging_operator_namespace

File: tests/internal/config/config.go
Comment (kononovn) [inline tests/internal/config/config.go:33]: LoggingOperatorNamespace
ECO_LOGGING_OPERATOR_NAMESPACE

**Review by kononovn — CHANGES_REQUESTED** [review]
Please check few comments

**Comment by klaskosk** [comment]
This passed the tests for the RAN team so lgtm once Nikita's comments are addressed

File: tests/internal/params/vars.go
Comment (edcdavid) [reply tests/internal/params/vars.go]: Fixed. I added a variable here before and reformatted.

File: tests/internal/config/default.yaml
Comment (edcdavid) [reply tests/internal/config/default.yaml]: Fixed

File: tests/internal/config/config.go
Comment (edcdavid) [reply tests/internal/config/config.go]: Fixed

**Comment by edcdavid** [comment]
@klaskosk I moved the ExecLocalCommand back to RAN area after a comment from @kononovn . I did not recreate the cluster.go, instead I moved it to the ranhelper, since this is running commands locally

File: tests/internal/config/config.go
Comment (kononovn) [inline tests/internal/config/config.go:33]: Please change LoggingNamespace > LoggingOperatorNamespace

File: tests/internal/cluster/cluster.go
Comment (kononovn) [inline tests/internal/cluster/cluster.go:191]: It looks like wait PollImmediate* pattern. Please use wait functions instead of sleep

File: tests/internal/cluster/cluster.go
Comment (kononovn) [inline tests/internal/cluster/cluster.go:217]: It looks like wait PollImmediate* pattern. Please use wait functions instead of sleep

**Review by kononovn — CHANGES_REQUESTED** [review]
Global level looks good. Just few comments to fix.

File: tests/internal/config/config.go
Comment (edcdavid) [reply tests/internal/config/config.go]: Fixed

File: tests/internal/cluster/cluster.go
Comment (edcdavid) [reply tests/internal/cluster/cluster.go]: Fixed

File: tests/internal/cluster/cluster.go
Comment (edcdavid) [reply tests/internal/cluster/cluster.go]: Fixed

**Review by kononovn — APPROVED** [review]
global level 
/lgtm

**Review by klaskosk — APPROVED** [review]
/lgtm for ran

**Pull request merged**
